bug: make test fails with a status code != 0

Dear all,

I've set up a developing environment as described [here](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md#setting-up-a-development-environment), but when doing the `make test` phase, I obtain an error message like this:
```
----------------------------------------------------------------------
Ran 2502 tests in 49.091s

OK
flake8 skbio setup.py checklist.py
./checklist.py
./checklist.py:215: ChecklistWarning: File skbio/metadata/tests/test_intersection.py has IGNORE directive. Ignoring scikit-bio copyright header validation.
  ChecklistWarning)
The following tests import `A` but should import `B` (file: A => B):
    skbio/alignment/tests/test_pairwise.py: skbio.util._decorator.classproperty => skbio.util.classproperty
    skbio/alignment/tests/test_tabular_msa.py: skbio.util._decorator.classproperty => skbio.util.classproperty
    skbio/alignment/tests/test_tabular_msa.py: skbio.util._testing.assert_data_frame_almost_equal => skbio.util.assert_data_frame_almost_equal
    skbio/io/format/tests/test_fastq.py: skbio.sequence._grammared_sequence.GrammaredSequence => skbio.sequence.GrammaredSequence
    skbio/io/format/tests/test_fastq.py: skbio.util._decorator.classproperty => skbio.util.classproperty
    skbio/io/format/tests/test_clustal.py: skbio.sequence._grammared_sequence.GrammaredSequence => skbio.sequence.GrammaredSequence
    skbio/io/format/tests/test_clustal.py: skbio.util._decorator.classproperty => skbio.util.classproperty
    skbio/io/format/tests/test_fasta.py: skbio.sequence._grammared_sequence.GrammaredSequence => skbio.sequence.GrammaredSequence
    skbio/io/format/tests/test_fasta.py: skbio.util._decorator.classproperty => skbio.util.classproperty

Makefile:23: set di istruzioni per l'obiettivo "test" non riuscito
make: *** [test] Errore 1
```
The error is thrown by `./checklist.py`, which raises this warnings but returns a `1` status code and test fails. I noticed that classes like `skbio.util._decorator.classproperty` are imported from the main module `skbio.util` so I've fixed imports to have a `0` status from `./checklist.py`. I will try to submit those fixes with a pull request. Maybe this problems is related in a different version of a module required by your test. [Here](https://gist.github.com/bunop/8e279640417c5cd1a5fb1759b6f4201c) are details of my conda environment.

Regards,

Paolo

--- a/skbio/alignment/tests/test_pairwise.py
+++ b/skbio/alignment/tests/test_pairwise.py
@@ -22,7 +22,8 @@ from skbio.alignment._pairwise import (
     _compute_score_and_traceback_matrices, _traceback, _first_largest,
     _compute_substitution_score)
 from skbio.sequence import GrammaredSequence
-from skbio.util._decorator import classproperty, overrides
+from skbio.util import classproperty
+from skbio.util._decorator import overrides
 
 
 class CustomSequence(GrammaredSequence):

--- a/skbio/alignment/tests/test_tabular_msa.py
+++ b/skbio/alignment/tests/test_tabular_msa.py
@@ -20,12 +20,13 @@ import scipy.stats
 
 from skbio import Sequence, DNA, RNA, Protein, TabularMSA
 from skbio.sequence import GrammaredSequence
-from skbio.util._decorator import classproperty, overrides
+from skbio.util import classproperty
+from skbio.util._decorator import overrides
 from skbio.util._testing import ReallyEqualMixin
 from skbio.metadata._testing import (MetadataMixinTests,
                                      PositionalMetadataMixinTests)
-from skbio.util._testing import (assert_index_equal,
-                                 assert_data_frame_almost_equal)
+from skbio.util import assert_data_frame_almost_equal
+from skbio.util._testing import assert_index_equal
 
 
 class TabularMSASubclass(TabularMSA):

--- a/skbio/io/format/tests/test_clustal.py
+++ b/skbio/io/format/tests/test_clustal.py
@@ -11,8 +11,9 @@ from io import StringIO
 from unittest import TestCase, main
 
 from skbio import TabularMSA
-from skbio.sequence._grammared_sequence import GrammaredSequence
-from skbio.util._decorator import classproperty, overrides
+from skbio.sequence import GrammaredSequence
+from skbio.util import classproperty
+from skbio.util._decorator import overrides
 from skbio.io.format.clustal import (
     _clustal_to_tabular_msa, _tabular_msa_to_clustal, _clustal_sniffer,
     _is_clustal_seq_line, _delete_trailing_number, _check_length,

--- a/skbio/io/format/tests/test_fasta.py
+++ b/skbio/io/format/tests/test_fasta.py
@@ -22,9 +22,10 @@ from skbio.io.format.fasta import (
     _fasta_to_tabular_msa, _generator_to_fasta,
     _sequence_to_fasta, _dna_to_fasta, _rna_to_fasta, _protein_to_fasta,
     _tabular_msa_to_fasta)
-from skbio.sequence._grammared_sequence import GrammaredSequence
+from skbio.sequence import GrammaredSequence
 from skbio.util import get_data_path
-from skbio.util._decorator import classproperty, overrides
+from skbio.util import classproperty
+from skbio.util._decorator import overrides
 
 
 class CustomSequence(GrammaredSequence):

--- a/skbio/io/format/tests/test_fastq.py
+++ b/skbio/io/format/tests/test_fastq.py
@@ -17,9 +17,10 @@ from skbio.io import FASTQFormatError
 from skbio.io.format.fastq import (
     _fastq_sniffer, _fastq_to_generator, _fastq_to_tabular_msa,
     _generator_to_fastq, _tabular_msa_to_fastq)
-from skbio.sequence._grammared_sequence import GrammaredSequence
+from skbio.sequence import GrammaredSequence
 from skbio.util import get_data_path
-from skbio.util._decorator import classproperty, overrides
+from skbio.util import classproperty
+from skbio.util._decorator import overrides
 
 import numpy as np
 

